src/pages/library/Showcase.js

```diff
@@ -16,104 +18,333 @@ import * as Style from 'src/libs/style'
+    ]),
+    div({
+      style: { flex: 'none' }
+    }, [h(RCollapse, { isOpened }, [children])])
+  ])
+}
+
+const groupByFeaturedTags = workspaces => _.flow([
+  _.map(tag => [tag, _.filter(w => _.includes(tag, w.tags.items), workspaces)]),
+  _.fromPairs
+])(uniqueSidebarTags)
+
+const Pill = ({ count, highlight }) => {
+  return div({
+    style: _.merge(styles.pill, highlight ? styles.pillHighlight : {})
+  }, [count])
+}
+
+const Sidebar = ({ onSectionFilter, onTagFilter, sections, selectedSections, selectedTags, workspacesByTag }) => {
+  const [collapsedSections, setCollapsedSections] = useState([])
+
+  const unionSectionWorkspaces = section => {
+    return _.uniq(_.flatMap(tag => workspacesByTag[tag], section.tags))
+  }
+
+  return div({ style: { display: 'flex', flexDirection: 'column' } }, [
+    _.map(section => {
+      return section.keepCollapsed ?
+        h(Clickable, {
+          key: section.name,
+          onClick: () => onSectionFilter(section),
+          style: { ...styles.sidebarRow, ...styles.nav.navSection }
+        }, [
+          div({ style: { flex: 1 } }, [section.name]),
+          h(Pill, {
+            count: _.size(unionSectionWorkspaces(section)),
+            highlight: _.includes(section, selectedSections)
+          })
+        ]) :
+        h(SidebarCollapser,
+          {
+            key: section.name,
+            title: section.name,
+            onClick: () => setCollapsedSections(_.xor([section], collapsedSections)),
+            isOpened: !_.includes(section, collapsedSections)
+          },
+          [
+            ..._.map(label => {
+              const tag = _.toLower(label)
+              return h(Clickable, {
+                style: { display: 'flex', alignItems: 'baseline', margin: '0.5rem 0' },
+                onClick: () => onTagFilter(tag)
+              }, [
+                div({ style: { flex: 1 } }, [label]),
+                h(Pill, {
+                  count: _.size(workspacesByTag[tag]),
+                  highlight: _.includes(tag, selectedTags)
+                })
+              ])
+            }, section.labels)
+          ]
+        )
+    }, sections)
+  ])
+}
 
 const Showcase = () => {
-  const { workspaces } = useWorkspaces()
   const stateHistory = StateHistory.get()
   const [featuredList, setFeaturedList] = useState(stateHistory.featuredList)
+  const [workspacesByTag, setWorkspacesByTag] = useState({})
+  const [sections, setSections] = useState([])
+
+  const [selectedSections, setSelectedSections] = useState([])
+  const [selectedTags, setSelectedTags] = useState([])
+  const [searchFilter, setSearchFilter] = useState()
+  const [sort, setSort] = useState('most recent')
+
+  const sortWorkspaces = Utils.cond(
+    [sort === 'most recent', () => _.orderBy(['created'], ['desc'])],
+    [sort === 'alphabetical', () => _.orderBy(w => _.toLower(_.trim(w.name)), ['asc'])],
+    () => _.identity
```

Comment:
I don't see a need for this default case since there's no way for this to be something else; that said, I'd tell you to use a switchcase instead of a cond, but with only two options could you just use a ternary?

Reply:
Yeah, I definitely agree with the `switchCase`, and you're right that the default case isn't needed. I do want to stop short of the ternary; neither feels like a natural primary or secondary case to me.